sycl/include/CL/sycl/atomic_ref.hpp

```diff
@@ -0,0 +1,668 @@
+//==----- atomic_ref.hpp - SYCL 2020 atomic_ref ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/__spirv/spirv_ops.hpp>
+#include <CL/sycl/access/access.hpp>
+#include <CL/sycl/atomic.hpp>
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/detail/spirv.hpp>
+#include <CL/sycl/memory_enums.hpp>
+#ifdef __SYCL_DEVICE_ONLY__
+#include <CL/sycl/multi_ptr.hpp>
+#endif
+
+#ifndef __SYCL_DEVICE_ONLY__
+#include <atomic>
+#endif
+#include <type_traits>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+using memory_order = cl::sycl::memory_order;
+using memory_scope = cl::sycl::memory_scope;
+
+#if __cplusplus < 201703L
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_relaxed =
+    memory_order::relaxed;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_acquire =
+    memory_order::acquire;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_release =
+    memory_order::release;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_acq_rel =
+    memory_order::acq_rel;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_seq_cst =
+    memory_order::seq_cst;
+#endif
+
+template <typename T>
+using IsValidAtomicRefType =
+    bool_constant<IsValidAtomicType<T>::value || std::is_pointer<T>::value>;
```

Comment:
This will have to change when `double` drops out of `IsValidAtomicType`.

Reply:
@Pennycook Yes, I've changed both definitions of `IsValidAtomicRefType`, in the `sycl::` as well as in `sycl::ext::oneapi::` namespaces.